test/integration/regression/test_regression_bug335.py

```diff
@@ -0,0 +1,85 @@
+import sys
+from pathlib import Path
+
+import pytest
+from sqlalchemy import (
+    Column,
+    Integer,
+    MetaData,
+    String,
+    Table,
+    create_engine,
+    insert,
+)
+
+PROJECT_ROOT = Path(__file__).parent / ".." / ".." / ".."
+SCRIPTS = PROJECT_ROOT / "scripts"
+ODBC_DRIVER = PROJECT_ROOT / "driver" / "libexaodbc-uo2214lv2.so"
+sys.path.append(f"{SCRIPTS.resolve()}")
+
+
+@pytest.fixture
+def pyodbc_connection_string(exasol_config):
+    config = exasol_config
+    return (
+        f"exa+pyodbc://{config.username}:{config.password}@{config.host}:{config.port}/"
+        f"?DEFAULTPARAMSIZE=200&INTTYPESINRESULTSIFPOSSIBLE=y"
+        "&FINGERPRINT=NOCERTCHECK&CONNECTIONLCALL=en_US.UTF-8&driver=EXAODBC"
+    )
+
+
+@pytest.fixture()
+def test_schema(control_connection):
+    connection = control_connection
+    schema = "REGRESSION_335"
+    connection.execute(f"CREATE SCHEMA {schema}")
+    connection.commit()
+    yield schema
+    connection.execute(f"DROP SCHEMA IF EXISTS {schema} CASCADE")
+    connection.commit()
+
+
+@pytest.fixture()
+def users_table(control_connection, test_schema):
+    connection = control_connection
+    table_name = "users"
+    connection.execute(
+        f"create table {test_schema}.{table_name} (id DECIMAL(18) identity primary key, name VARCHAR(2000) UTF8)"
+    )
+    connection.commit()
+    yield test_schema, table_name
+
+
+def test_lastrowid_does_not_create_extra_commit(
+    exasol_config, users_table, pyodbc_connection_string
+):
+    """
+    For further details on this regression see `Issue-335 <https://github.com/exasol/sqlalchemy-exasol/issues/335>`_.
+    """
+    schema_name, table_name = users_table
+    metadata = MetaData()
+    engine = create_engine(pyodbc_connection_string)
+
+    table = Table(
+        table_name,
+        metadata,
+        Column("id", Integer, primary_key=True),
+        Column("name", String(2000)),
+        schema=schema_name,
+    )
+
+    from odbc import odbcconfig
+
+    with odbcconfig(ODBC_DRIVER):
+        conn = engine.connect()
+        trans = conn.begin()
+
+        # insert without and explicit id will trigger get_lastrowid
+        insert_statement = insert(table).values(name="Gandalf")
+        conn.execute(insert_statement)
```

Comment:
I am guessing here is where the implicit commit happened before. Maybe you should add a comment here.